include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -624,14 +624,26 @@ distribute_local_to_global (const Vector<LocalType>      &local_vector,
                             VectorType                   &global_vector,
                             const FullMatrix<LocalType>  &local_matrix) const
 {
+  distribute_local_to_global(local_vector,local_dof_indices,local_dof_indices, global_vector, local_matrix);
+}
+
+template <typename VectorType, typename LocalType>
+void
+ConstraintMatrix::
+distribute_local_to_global (const Vector<LocalType>      &local_vector,
+                            const std::vector<size_type> &local_dof_indices_row,
+                            const std::vector<size_type> &local_dof_indices_col,
+                            VectorType                   &global_vector,
+                            const FullMatrix<LocalType>  &local_matrix) const
+{
   Assert (sorted == true, ExcMatrixNotClosed());
-  AssertDimension (local_vector.size(), local_dof_indices.size());
-  AssertDimension (local_matrix.m(), local_dof_indices.size());
-  AssertDimension (local_matrix.n(), local_dof_indices.size());
+  AssertDimension (local_vector.size(), local_dof_indices_col.size());
+  AssertDimension (local_matrix.m(), local_dof_indices_row.size());
+  AssertDimension (local_matrix.n(), local_dof_indices_col.size());
 
   const size_type n_local_dofs = local_vector.size();
   if (lines.empty())
-    global_vector.add(local_dof_indices, local_vector);
+    global_vector.add(local_dof_indices_col, local_vector);
```

Comment:
To me, it looks like the rows and columns are the wrong way around. This function is supposed to assemble a right hand side vector, so in the usual matrix notation the indices you use for accessing the global vector should be the `local_dof_indices_row`, isn't it?